contracts/vesting-lp/src/contract.rs

```diff
@@ -64,31 +64,11 @@ pub fn execute(
 pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
     let vest_app = BaseVesting::new(Strategy::EveryBlock);
     match msg {
-        QueryMsg::Config {} => vest_app.query(deps, env, QueryBase::Config {}),
-        QueryMsg::VestingAccount { address } => {
-            vest_app.query(deps, env, QueryBase::VestingAccount { address })
-        }
-        QueryMsg::VestingAccounts {
-            start_after,
-            limit,
-            order_by,
-        } => vest_app.query(
-            deps,
-            env,
-            QueryBase::VestingAccounts {
-                start_after,
-                limit,
-                order_by,
-            },
-        ),
-        QueryMsg::AvailableAmount { address } => {
-            vest_app.query(deps, env, QueryBase::AvailableAmount { address })
-        }
-        QueryMsg::Timestamp {} => vest_app.query(deps, env, QueryBase::Timestamp {}),
-        QueryMsg::UnclaimedAmountAtHeight { address, height } => Ok(to_binary(
+        QueryMsg::Base(base_query) => vest_app.query(deps, env, base_query),
+        QueryMsg::Ext(ExtraQueryMsg::UnclaimedAmountAtHeight { address, height }) => Ok(to_binary(
             &query_unclaimed_amount_at_height(&vest_app, deps, address, height)?,
         )?),
-        QueryMsg::UnclaimedTotalAmountAtHeight { height } => Ok(to_binary(
+        QueryMsg::Ext(ExtraQueryMsg::UnclaimedTotalAmountAtHeight { height }) => Ok(to_binary(
```

Comment:
will there be vesting vault later that will use these `unclaimed_at_height` queries in its own `power_at_height` queries?

Reply:
discussed that this contract doesn't have to comply with the vault interface because there will be a specific vault contract which will call these queries in its own `power_at_height` handlers